### PR TITLE
extended form item schema

### PR DESCRIPTION
**Pull request by mvnm**

### CODE REVIEW

File: docs/building/forms.md [hunk 1]
```diff
@@ -21,36 +21,61 @@ Form
...
+        "datetime"  - the user should enter a date and time
+        "form-menu" - the user should choose an option from the menu
+
+    name                       - (string) the name of this FormItem, used in form serialization
+
+    body                       - (array of objects) MenuItemFormItem objects - see below - required only for type=form-menu
+    chunking_footer            - (string) shown in the footer of the sms chunks
+    confirmation_label         - (string) shown in the confirmation menu
+    description                - (string) the description of this FormItem
+    editable                   - (bool) indicates this form item is editable or not
```
Comment (romeo1m) [inline docs/building/forms.md]: From my point of view this field should be the reverse of hidden. I mean, a hidden item should be not editable. Too many options lead to even more combinations of them that can be hard to explain or to be understood. What do you think?

Comment (mvnm) [reply docs/building/forms.md]: Yes, you are right. It's redundant https://github.com/ONEmCommunications/nautilus/commit/9e40cce93#diff-e31d0443f9ab445b7b94b64ed3b628a2R320

Comment (mvnm) [reply docs/building/forms.md]: Probably better to just take it out ..

File: docs/building/forms.md [hunk 2]
```diff
@@ -21,36 +21,61 @@ Form
...
+    max_length_error           - (string) message to be shown on max_length error
+    max_value                  - (int) validates the user input - for type=int|float
+    max_value_error            - (string) message to be shown on max_value error
+    meta                       - (object) MenuFormItemMeta object - see below - applies only for type=form-menu
+    method                     - (string) http method, how the callback url should be triggered
+    min_length                 - (int) validates the user input - for type=string
+    min_length_error           - (string) message to be shown on min_length error
+    min_value                  - (int) validates the user input - for type=int|float
+    min_value_error            - (string) message to be shown on min_value error
+    required                   - (bool) can be skipped if set to false
```
Comment (romeo1m) [inline docs/building/forms.md]: This means we support SKIP command, or how does a not required step look like?

Comment (mvnm) [reply docs/building/forms.md]: Yes, already supported by the wizard engine ..

File: docs/building/forms.md [hunk 3]
```diff
@@ -21,36 +21,60 @@ Form
...
+    validate_url               - (string) the callback url path (GET) triggered to validate user input with query string ?name=user_input - url must return json content {'valid': True/False, 'error': 'Some validation error message'}
+    value                      - (string) required for type=hidden
+    
+
+MenuItemFormItem
+    type - (string) enum: "option", "content"
+        "option"  - needs a value
+        "content" - does not need a value - shows content as indicated in description
+
+    value       - (string) the value of this FormItemMenuItem, used in form serialization
```
Comment (romeo1m) [inline docs/building/forms.md]: You renamed this to `MenuItemFormItem`. See also the line below.

Comment (mvnm) [reply docs/building/forms.md]: Well spotted! Renamed here https://github.com/ONEmCommunications/documents/pull/12/commits/849d5a0b14049a9d6e2e50ba65c31d95c97b2fae

**Pull request merged**